### PR TITLE
DCOS-41250: implement basic soak tests for DC/OS UI

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

That's pretty hard, you can run the script against a cluster by setting the environment variables (according to the readme, therefore I won't give you a command). 

## Trade-offs

- Decided to use puppeteer (which has a high level of abstraction) to move fast. We might need to move to sth that supports multiple browsers later, but first starting to make experiences with soaking in general is a good first step.
- Decided to manually log in, partly because I think our test should fail if the login fails, also because I think adding some python utils to handle token generation won't help wiht maintainability

## Dependencies

This PR needs to be merged before we can deploy it to soak.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -92,6 +92,22 @@ pipeline {
...
+    stage("Soak Docker Release") {
+      agent {
+        label "mesos-ubuntu"
+      }
+      steps {
+        withCredentials([
+          string(credentialsId: '7bdd2775-2911-41ba-918f-59c8ae52326d', variable: 'DOCKER_HUB_USERNAME'),
+          string(credentialsId: '42f2e3fb-3f4f-47b2-a128-10ac6d0f6825', variable: 'DOCKER_HUB_PASSWORD')
+        ]) {
+          sh "docker --help"
```
Comment (nLight) [inline Jenkinsfile]: we don't really need this, do we?

File: soak-tests/soak112s/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+set -ex
+cd "${0%/*}" # to the folder containing this script.
+
+IMAGE_IDENTIFIER='mesosphere/dcos-ui-soak'
```
Comment (nLight) [inline soak-tests/soak112s/deploy.sh]: Shall we add label to distinguish between soak versions? I.e. 'mesosphere/dcos-ui-soak:112s'?

**Comment by nLight** [comment]
Just crossed my mind. Having it in the OSS repo would mean we don't soak EE plugins, thus it won't be able to login, right?

File: soak-tests/soak112s/deploy.sh (hunk 2 above)
Comment (DanielMSchmidt) [reply soak-tests/soak112s/deploy.sh]: Good point, will do

File: Jenkinsfile (hunk 1 above)
Comment (DanielMSchmidt) [reply Jenkinsfile]: We do as long as I debug why we can not push to dockerhub :D

**Comment by DanielMSchmidt** [comment]
@nLight you are right, we should move this 🤔 I'll make it work and move it then

**Comment by nLight** [comment]
Even with the special Jenkins node it doesn't work?

**Comment by DanielMSchmidt** [comment]
@nLight Nope, couldn't get it running on any node so far. @juliangieseke mentioned that Docker in Docker could also referr to being able to use a different dockerfile on these agents and not promise a docker deamon to be available. We briefly thought about installing docker on the mac mini and calling it a day, but this seems like a non-ideal solution.

**Comment by nLight** [comment]
Shall we move the code into EE repo and scope out the CI part? 
So that before another SOAK test we would just manually build the image? (< Should be documented)

**Comment by nLight** [comment]
I bet we could just extract this step and setup a separate job that would run in `docker-in-docker` image. But I'd say let's scope that out

**Comment by DanielMSchmidt** [comment]
@nLight I tried out 4 different jenkins labels with docker in docker and neither of them worked. I can move it over and we scope it out, seems reasonable to me.

**Pull request closed without merging**